apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,198 @@
+import { EncodingType, PriceFeedMetadata, PriceUpdate } from "./types";
+import EventSource from "eventsource";
+
+const DEFAULT_TIMEOUT: DurationInMs = 5000;
+const DEFAULT_HTTP_RETRIES = 3;
+
+export type UnixTimestamp = number;
+export type DurationInSeconds = number;
+export type HexString = string;
+export type DurationInMs = number;
+
+export type HermesConnectionConfig = {
+  /* Timeout of each request (for all of retries). Default: 5000ms */
+  timeout?: DurationInMs;
+  /**
+   * Number of times a HTTP request will be retried before the API returns a failure. Default: 3.
+   *
+   * The connection uses exponential back-off for the delay between retries. However,
+   * it will timeout regardless of the retries at the configured `timeout` time.
+   */
+  httpRetries?: number;
+};
+
+export class HermesConnection {
```

Comment:
Minor nit but IMO there's not much of a good reason to make this a class.  There isn't any internal state being tracked, and form a PLT perspective, state management is really the only thing that classes actually exist for.

This is personal preference but for things that aren't stateful, I prefer doing this:

```tsx
import { doAThing } from "hermes-client";

const sharedOptions = { ... };

doAThing({ ...sharedOptions, ...moreOptions });
```

Even though it's is more verbose, I actually greatly prefer it.  There's much less indirection and it's much easier to follow what's happening at the call site, rather than trying to find options that get passed in to some constructor somewhere which may be far away and may not be obvious at the call site actually have influence over the behavior.

This is totally opinion and feel free to disregard.